packages/ccd-js-gen/README.md

```diff
@@ -0,0 +1,484 @@
+
+#### function `parseEvent`
+
+Parse a raw contract event logged by this contract into a structured representation.
+
+_This is only generated when the schema contains contract event type._
+
+**Parameter:** `event` The contract event to parse.
+
+**Returns:** The structured event of the `Event` type (see type above).
+
+```typescript
+const rawContractEvent = ...; // The unparsed contract event from some transaction.
+const event: MyContract.Event = MyContract.parseEvent(rawContractEvent);
+```
+
+#### type `<EntrypointName>Parameter`
+
+Type representing the parameter of for an entrypoint. The type is named `<EntrypointName>Parameter` where `<EntrypointName>` is the name of the entrypoint in Pascal case.
+
+_This is only generated when the schema contains contract entrypoint parameter type._
+
+#### function `send<EntrypointName>`
+
+For each entrypoint of the smart contract a function for sending a transaction calling this entrypoint is produced.
+These are named `send<EntrypointName>` where `<EntrypointName>` is the name of the entrypoint in Pascal case.
+
+An example for a smart contract with an entrypoint named `launch-rocket`, the function becomes `sendLaunchRocket`.
+
+**Parameters**
+
+The function parameters are:
+
+- `contractClient` The client of the smart contract instance.
+- `transactionMetadata` Metadata related to constructing the transaction, such as energy and CCD amount to include.
+- `parameter` Parameter to provide to the smart contract entrypoint.
+
+  _With schema type:_ If the schema contains type information for the parameter, a type for the parameter is generated and used for this function (see type above).
+
+  _Without schema type:_ If no schema information is present, the function uses the generic `Parameter` from `@concordium/web-sdk`.
+
+- `signer` The keys of to use for signing the transaction.
+
+**Returns**: Promise with the hash of the transaction.
+
+```typescript
+const myContract: MyContract.Type = ...; // Generated contract client.
+const transactionMeta = { ... }; // Transaction meta information.
+const parameter = ...; // Parameter to pass the smart contract entrypoint.
+const signer = ...; // The keys to use for signing the transaction.
+const transactionHash = await MyContract.sendLaunchRocket(myContract, transactionMeta, parameter, signer);
+```
+
+#### function `dryRun<EntrypointName>`
+
+For each entrypoint of the smart contract a function for dry-running a transaction calling this entrypoint is produced.
+These are named `dryRun<EntrypointName>` where `<EntrypointName>` is the name of the entrypoint in Pascal case.
+
+An example for a smart contract with an entrypoint named `launch-rocket`, the function becomes `dryRunLaunchRocket`.
+
+**Parameters**
+
+The function parameters are:
+
+- `contractClient` The client of the smart contract instance.
+- `invoker` TODO
+- `parameter` Parameter to provide to the smart contract entrypoint.
```

Comment:
What about amount?

Reply:
It now takes metadata object similar to when sending a transaction, this provides a way to provide the amount, invoker and energy.